controllers/workloads/console/controller.go

```diff
@@ -592,6 +653,126 @@ func requeueAfterInterval(logger logr.Logger, interval time.Duration) reconcile.
 	return reconcile.Result{Requeue: true, RequeueAfter: interval}
 }
 
+func sessionRecordFileName(ix int) string {
+	return fmt.Sprintf("%s/output%0d", SessionRecVolMount, ix)
+}
+
+func (r *ConsoleReconciler) buildSidecarContainer() corev1.Container {
+	return corev1.Container{
+		Name:            "session-streamer",
+		Image:           r.SessionSidecarImage,
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      SessionRecVolName,
+				MountPath: SessionRecVolMount,
+				ReadOnly:  false,
+			},
+		},
+		Resources: corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{
+				corev1.ResourceMemory: resource.MustParse("512Mi"),
+				corev1.ResourceCPU:    resource.MustParse("512m"),
+			},
+			Requests: corev1.ResourceList{
+				corev1.ResourceMemory: resource.MustParse("512Mi"),
+				corev1.ResourceCPU:    resource.MustParse("512m"),
+			},
+		},
+		Env: []corev1.EnvVar{
+			{
+				Name: "KUBERNETES_POD_NAME",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						FieldPath: "metadata.name",
+					},
+				},
+			},
+			{
+				Name: "KUBERNETES_NAMESPACE",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						FieldPath: "metadata.namespace",
+					},
+				},
+			},
+			{
+				Name:  "SIDECAR_SHUTDOWN_DELAY",
+				Value: fmt.Sprint(SidewrapShutdownDelay),
+			},
+			{
+				Name:  "SIDECAR_GRACE_PERIOD",
+				Value: fmt.Sprint(SidewrapGracePeriod),
+			},
+			{
+				Name:  "TOPIC_ID",
+				Value: r.SessionPubsubTopicId,
+			},
+			{
+				Name:  "PROJECT_ID",
+				Value: r.SessionPubsubProjectId,
+			},
+		},
+	}
+}
+
+func (r *ConsoleReconciler) modifyPodTemplate(logger logr.Logger, podTemplate *corev1.PodTemplateSpec) *corev1.PodTemplateSpec {
+
+	mutatedTemplate := podTemplate.DeepCopy()
+
+	// Add volume mount for session recording data
+	mutatedTemplate.Spec.Volumes = append(
+		mutatedTemplate.Spec.Volumes,
+		corev1.Volume{
+			Name: SessionRecVolName,
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+	)
+
+	// Modify each container command and args to start the
+	// session recording wrapper which spawns the original
+	// command
+	for ix := range mutatedTemplate.Spec.Containers {
+		ctr := &mutatedTemplate.Spec.Containers[ix]
+		execCommand := []string{"--"}
+		execCommand = append(execCommand, ctr.Command...)
+		execCommand = append(execCommand, ctr.Args...)
+
+		ctr.Command = []string{"tlog-rec"}
+		args := []string{"-o", sessionRecordFileName(ix)}
```

Comment:
I think we may also want to specify `--log-input=true`, so that we capture stdin separately (the default is false)

Reply:
Great catch, thanks